### DISCUSSION TITLE
Can I make the tour progress match only visible targets?

### DISCUSSION THREAD

**Discussion by NazarPo** [gilbarbara/react-joyride#896] [Q&A] [answered]
Hi. I'm looking for a straightforward solution to the following problem. Say, I have a permission gate in place that does not render a button if a user does not have required privileges. Yet, the button (if it's rendered) is a target for one of the tour steps. Currently,

- if a user has all privileges required by tour steps, I can see all steps
- if a user does not have a certain privilege to one target (say "Delete" button), the step gets skipped, and the tour gets moved further to the next step. This works just as expected.

But here is the problem. Say, I have 5 targets (i.e. 5 steps) on the page. The "Delete" button, for instance, is the 3rd target, and it's not visible (I have no privileges to see and use it). When I click "Next", the tour jumps from the 2nd step straight to the 4th, thus showing me the progress like this `4 / 5`. This may confuse users, because they haven't seen the 3rd step. And the total, which is `5`, does not match the actual number of visible targets.

**The question is** if it's possible to override this behaviour so that I can see only 4 steps in the tour progress if one target is not visible (rendered) **without turning Joyride into `controlled` mode.**

P.S. I'm pretty sure this problem can be resolved by turning Joyride into `controlled` mode and assigning privileges to each step corresponding to their targets. But, I'd like to omit this if possible for two reasons:
1. I'll have to cover the `controlled` logic with unit tests myself
2. I'd like to make the tour as independent as possible. If one adds/removes privilege to/from targets, but forgets to do the same with the tour steps, that would produce a bug

**Comment by gilbarbara** [comment, marked as answer]
Hey @NazarPo 

The only option I see is not to include the step if the user cannot view it.

**Reply by NazarPo** [reply]
Hi @gilbarbara. I see. I'll figure something out then. Anyway, thanks for looking into this and the immediate response.
